### ISSUE TITLE
ReplayGain values are totally wrong

### ISSUE THREAD

**Issue by breversa** [vanilla-music/vanilla#701]
- Vanilla Music Version (Settings -> About): 1.0.57
- Android Version: 6.0.1
- Phone vendor / model: LG G5 H815
- Application was downloaded from (use `[x]`):
  - [ ] Play Store
  - [x] F-Droid
  - [ ] Other (Nightly Build)
- Did this work in a previous version?
  - [ ] Yes
  - [ ] No
  - [X] NA / Don't know.

## Description
The dispalyed ReplayGain values are totally wrong, and vary according to the file type :
- FLAC seems hard-set to +5dB (for both track and album gain)
- Ogg Vorbis is either +5db OR 0dB OR correct (for album gain only)
- MP3 is just wrong

## Steps to reproduce the issue
1. Calculate ReplayGain with your favorite tool (I've been using SoundKonverter, and recently moved to the ReplayGain Beets plugin).
2. Check the calculated gain with your favorite tool (e.g. SoundKonverter).
3. Play the files in Vanilla Music : the displayed ReplayGain values are totally different, and the sound volume is ajusted accordingly.

## Screenshot
Expected result :
![snapshot2](https://user-images.githubusercontent.com/2071276/33566292-c8c15bce-d91f-11e7-8aa2-321317b53820.png)

Current result :
FLAC file :
![screenshot_20171130-223032](https://user-images.githubusercontent.com/2071276/33566068-373b9ffc-d91f-11e7-9e8e-64c192f84466.png)

Ogg Vorbis files :
![screenshot_20171130-223002](https://user-images.githubusercontent.com/2071276/33566080-3fb8b070-d91f-11e7-84a8-94169eecedc3.png)
![screenshot_20171130-223052](https://user-images.githubusercontent.com/2071276/33566092-45b8f4d0-d91f-11e7-8510-28d611263cb5.png)
![screenshot_20171130-223114](https://user-images.githubusercontent.com/2071276/33566111-4d0d7be8-d91f-11e7-94ff-4d4b2c0fd6d4.png)
![screenshot_20171130-223147](https://user-images.githubusercontent.com/2071276/33566131-563519ba-d91f-11e7-8db9-598775d99f11.png)

MP3 files :
![screenshot_20171130-223101](https://user-images.githubusercontent.com/2071276/33566143-5b69f4a0-d91f-11e7-8081-2e84e3b407c3.png)
![screenshot_20171130-223459](https://user-images.githubusercontent.com/2071276/33566158-616f953a-d91f-11e7-91c0-9f89952f28e7.png)
...

Could it be a Vanilla Music internal library error, caused by multiple import/export of music ?

**Comment by adrian-bl** [comment]
Please post the output of

`metaflac --list $file`

**Comment by breversa** [comment]
```
metaflac --list Nightwish\ -\ Once\ upon\ a\ troubadour.flac 
METADATA block #0
  type: 0 (STREAMINFO)
  is last: false
  length: 34
  minimum blocksize: 4096 samples
  maximum blocksize: 4096 samples
  minimum framesize: 14 bytes
  maximum framesize: 13048 bytes
  sample_rate: 44100 Hz
  channels: 2
  bits-per-sample: 16
  total samples: 14147280
  MD5 signature: c467f704381bce6b09e88fcfc23bef86
METADATA block #1
  type: 3 (SEEKTABLE)
  is last: false
  length: 594
  seek points: 33
    point 0: sample_number=0, stream_offset=0, frame_samples=4096
    point 1: sample_number=438272, stream_offset=826392, frame_samples=4096
    point 2: sample_number=880640, stream_offset=1894730, frame_samples=4096
    point 3: sample_number=1318912, stream_offset=2978275, frame_samples=4096
    point 4: sample_number=1761280, stream_offset=4107961, frame_samples=4096
    point 5: sample_number=2203648, stream_offset=5187969, frame_samples=4096
    point 6: sample_number=2641920, stream_offset=6279431, frame_samples=4096
    point 7: sample_number=3084288, stream_offset=7359356, frame_samples=4096
    point 8: sample_number=3526656, stream_offset=8507456, frame_samples=4096
    point 9: sample_number=3964928, stream_offset=9659629, frame_samples=4096
    point 10: sample_number=4407296, stream_offset=10828932, frame_samples=4096
    point 11: sample_number=4849664, stream_offset=11947251, frame_samples=4096
    point 12: sample_number=5287936, stream_offset=12998026, frame_samples=4096
    point 13: sample_number=5730304, stream_offset=14055193, frame_samples=4096
    point 14: sample_number=6172672, stream_offset=15126011, frame_samples=4096
    point 15: sample_number=6610944, stream_offset=16216807, frame_samples=4096
    point 16: sample_number=7053312, stream_offset=17389773, frame_samples=4096
    point 17: sample_number=7495680, stream_offset=18615479, frame_samples=4096
    point 18: sample_number=7933952, stream_offset=19790960, frame_samples=4096
    point 19: sample_number=8376320, stream_offset=20993178, frame_samples=4096
    point 20: sample_number=8818688, stream_offset=22201905, frame_samples=4096
    point 21: sample_number=9256960, stream_offset=23424333, frame_samples=4096
    point 22: sample_number=9699328, stream_offset=24638449, frame_samples=4096
    point 23: sample_number=10141696, stream_offset=25823164, frame_samples=4096
    point 24: sample_number=10579968, stream_offset=26892572, frame_samples=4096
    point 25: sample_number=11022336, stream_offset=27964292, frame_samples=4096
    point 26: sample_number=11464704, stream_offset=29099536, frame_samples=4096
    point 27: sample_number=11902976, stream_offset=30262274, frame_samples=4096
    point 28: sample_number=12345344, stream_offset=31485701, frame_samples=4096
    point 29: sample_number=12787712, stream_offset=32659635, frame_samples=4096
    point 30: sample_number=13225984, stream_offset=33775704, frame_samples=4096
    point 31: sample_number=13668352, stream_offset=34769535, frame_samples=4096
    point 32: sample_number=14110720, stream_offset=35207543, frame_samples=4096
METADATA block #2
  type: 4 (VORBIS_COMMENT)
  is last: false
  length: 4683
  vendor string: reference libFLAC 1.3.0 20130526
  comments: 64
    comment[0]: FMPS_PLAYCOUNT=0
    comment[1]: FMPS_RATING=0.6
    comment[2]: FMPS_RATING_AMAROK_SCORE=0
    comment[3]: REPLAYGAIN_REFERENCE_LOUDNESS=89.0 dB
    comment[4]: ACOUSTID_FINGERPRINT=AQADtJ3SaFII-_iR7EesMKJwXjiOKQ8-Hc-Efsh6JGcujImOH6VEmFWMb0e6U8eVyNAXpEU_4sTD4xd-HF2Q8DnSpsE__IE5Pugi5XiHy6kWtCfCPC-Si7iE70F5-CmFP-iNY85HhCT0BZyl48cPRVOI69AehOGO58P1YFc-VOXQ4-GjIesv6EjPYHxRHk3S40ItJfCJw0_QLjti0YW2o2ma4Pvx449EnMqJHxf65WikucMbqNeGpimuKUf6rOg_9GiqF1f8YCiY3MeJiE6KH9dyfFJyKOoueBL0HD6-pci0KCh5Yc-FUEfjSNAVJfhxHc-FKkv4wFWLXkeOKz4eRseTExV5NA8eHjmS-UET88Ofo8mD_uiToelx3NIR6ocWEj9eifhTNAp6TceTZ3gchQp-HD8aJ4uL6slITJYOLbrALaKQJzvMiCnew3kV7MqFD6EZNH9w9Xie4g6GBvXxBWymB-V0OMbW47OOw00ylMTxSiiPpgl-_PAz_PiFJ6jh8TiHE64Y2Dz6BfULEd3G4zra44QuOBOe4eTx6DjhL6iJi4YsHpWOz4ST9jgqXDd-fEPv4ggvHm-iIA914vsD7YPHlshzoy-e4Tz6HKVEHPGLa1mOn8Px4UYf_Mcj44cTShKD0NKBHk6OHjGT8Jga0RBxUketLBi74_kDHZFoGc2ZGceP6dkLc-3R7DiO7g78HDmzI1mq4zouHj_gqSeeHF8iIj8NRYmPUD--u3iN6_BD_B9yjRLajLlwerh4oo-OUMnF40mCKXUYfOjh6mglNJWM8MlxEsfv4DJxoYd_9GmIRw6eHE0e9MUeZeibwxfaxQVz4hL2VEQf3EdJHq6DhoqUEbqfBN_RJ2g-qPkx8kHzKOGCPqNw5KTBG003KiHC58GHV_h0aGF0zDb8oS_wc9ClzAp644J34eiPh9vQTA7aBdcjdAzhJ8WzEv1xG5cWlNLhoxaNfgosXNPxnOh7PMJf-PpwGu9xNFVR8ziqSXgOh-LhCzcU6tFhTTnx7fjx4wuh7EqMGu9wPHqO49KJJ8uUCLdRw0e_4CMu4iwe63jT45zxQ_uHpk7w48yh5UeV45KYI-xyXC8wXSnRH02S6riOXkjeIB_6YLcT_Buq3GhGImycCcniHJeO5_Dx48NFigo6nYHOw_mOWktCHP1kOGaOLzpSaYk46PwQ6ituqvgC69cQ--gzND9mKeTRjiKaMYf6o3l0dCOu9UR7DeJzHLmWGi72oHmED_mOmxa00Y3gB53GxPjxYw976A-YHQ3VHFeO5gz6FeGHLw_x8yhCO7CTHYwFHc1R6saVK_iP_tCShFmIR5mPP3Hg49qEZqcwNbqC7wqaSqLQL6h7NB4uGw9p_LgEHdc5oVNeQU8GvDhOnIU_48GPE3BH3Mf1w1fwEeZR4heOxhN-_EVZBD9-_EfjFD3x4MfxRMiZ4NCXoV2iHOmIHz764_5QoemGazFOH0d4HuKHH7GDR0jFD7UtPMc1nHiM73iSBRX8FD0vHGEfGA754z58Hs-OaEd_6AkaaczxHIdTLbiEHvsu6EUzEldxZ7Dg8Lh-PDz6D0-OZhnRo0cYW8fR1EGVSJkQT5Fx_fglfEX5oumOaxWaPYdWouSRw0-M3_DRLsfxXEiWR0HY-JiP62jcgBcFJw0qHUce6ImQ-8GPm8bTo1TEIhf4ocmXUJAjP_h69DyaM_h0fBaq3aDk44euQ39hhscPz8fVQxQ19Mjx-Oh7_AG_waoiVJJxJUWuG7qL_EMtHc2z4xROoy-akwsefERDFZHkoD_Eo5N6hDseHSm5QNeR6viL4z08NSna4wwipd6IZMePZ0ezp6iVHe_wfaj05IGJXNqExsW1Hf3QfHKg7wF_XD_s6BgdJfjwo7oO_3A4d-hT3EdXwfnQt_iRH3vEB3k0gvNxNDq-PtiOHr8Qvjl8_PCCcMehxUOOZs_xTMePZ0f4QJx85K3h7yi17HiI48xRZngPh0P4MLgPHemCa8LbFL_Q50Qa56gd_EctwnuKnkc7GunKo8cVHD8Kb1mIT0eqo_oK_QjPwz8-Hu2PHv7wocZZ4YdnHPlanLISaMlRi3iiQ5Ne3Mhz4Tsu4UVeHv6hHR-HHyXDFCJxwQ-N97gLf0epXYSuHLUxhY-N62g-oU9xKhnILjgJ_YIdZksQND7B8Ec9hLmPWceJc9BR-Q107bg6HP0wHR9hDvX44DoeTPvh91DIKyh3vEeuHMwpGn58tMSRc0R1aEv2IXZigW_RVAz-FHP2oMSPPD3U50GjfBPi7fC0ibiLyTvy0bgW_bDmC-qaFk-3I1KO3rhUMcEf4dJ8NCoxiSby5LiOG308Qj-oS8fZXXik6RCrwlzWZOB1pJt21At6THl4XEZe4yl6iIea-cJFDbWIly5-PMdpfBkqEyaPB2WNH7XCoI9wHer84odL4UuOxzmaP0KvQ3_wHLvQTjoafvgBwDCghCCOGGCWCMEVY0AABCRjCApgyDFIEQEYIAAowg2iAAgljDLEAcyIgkIwIggjBAAIJDQGkCSIIAJ4AJBQihAiBCJCAasIAMYJiggRRhGEjDGSQAQsYcAQEZQACAhFjHBCECQJAcASQ5DABgFBlBGUAcOAM4gRgwRSEABAAJVKECeABYMiRRwwgBDEDBNECGecQ4wAJRwABDpAkXHSAScMAAYAwZRhjCkBsBAAKIuUQAAYAIjGGBFkgCAIuGQEEIQChIAwwCDCQAFIGeAYQAIogBARBhkGBEEAIHGpUYQBgwAkggGlDAACEQGEAAYQwIAgEDAgoBAEeUQIIg4QwAyhxChkjDEEGWMBMUYAIAgRACjgCBMEUCQIAQgjCgwACABACVMUEMeAEEAYpYQxgBCACBCICKMUAM4gAwSQlBDDBGICAgMMQCIQIgBBwBFgFQACGUMAIAwAkRBxhgENGPAGMEIEQkAJY4AwRnCgBCBEAgMI0EwwYggixBFGARDGgiYIU8oRQYAQilFgJRCAOAocEUJRaJkCQVCoFBCEMeEEIEQYRQU4VDCEhDFAAAWAYUApYoAEjGmALAIYCASYIgoxCyhBhCijmBJEkEMEIGIZ4RxCAIJFjJCEGaEcAkAAA5RyAkHhBBTKEUKAJYIAAQ0wgillDGCKEcCJAooAJwADCTiAnXNQGCMYEA4AQxgTChBDCBGCISGUAEgBQY0HQDBpLAFMOCIAJkIxKQARzDGgCHBCKESIQAABToAAADgKABEEKISgAMwhwoghBhkgiICOAEcIUEBYApEwSBACmKMEAGGAMEQggIwAAAiGiCICMMWBEUIcwgAUggCAFIOAGGIEQ9gJRwgTShggkHIGCCIYAIooAwiSSIAhGRAAAAGMooYAAC4wykhDARDGggAIAAQSZAQCAjoBrGcCMQ
    comment[5]: ACOUSTID_ID=
    comment[6]: ALBUM=Angels fall first
    comment[7]: ALBUM ARTIST=Nightwish
    comment[8]: ALBUMARTIST=Nightwish
    comment[9]: ALBUMARTIST_CREDIT=
    comment[10]: ALBUMARTISTSORT=
    comment[11]: MUSICBRAINZ_ALBUMCOMMENT=
    comment[12]: MUSICBRAINZ_ALBUMSTATUS=
    comment[13]: MUSICBRAINZ_ALBUMTYPE=
    comment[14]: ARRANGER=
    comment[15]: ARTIST=Nightwish
    comment[16]: ARTIST_CREDIT=
    comment[17]: ARTISTSORT=
    comment[18]: ASIN=
    comment[19]: BPM=0
    comment[20]: CATALOGNUMBER=
    comment[21]: DESCRIPTION=
    comment[22]: COMMENT=
    comment[23]: COMPILATION=0
    comment[24]: COMPOSER=
    comment[25]: COMPOSERSORT=
    comment[26]: RELEASECOUNTRY=
    comment[27]: DATE=1999
    comment[28]: YEAR=1999
    comment[29]: DISC=1
    comment[30]: DISCNUMBER=1
    comment[31]: DISCSUBTITLE=
    comment[32]: DISCTOTAL=1
    comment[33]: DISCC=1
    comment[34]: TOTALDISCS=1
    comment[35]: ENCODEDBY=
    comment[36]: ENCODER=
    comment[37]: GENRE=Heavy Metal
    comment[38]: GROUPING=
    comment[39]: LABEL=
    comment[40]: PUBLISHER=
    comment[41]: LANGUAGE=
    comment[42]: LYRICIST=
    comment[43]: LYRICS=
    comment[44]: MUSICBRAINZ_ALBUMARTISTID=
    comment[45]: MUSICBRAINZ_ALBUMID=
    comment[46]: MUSICBRAINZ_ARTISTID=
    comment[47]: MUSICBRAINZ_RELEASEGROUPID=
    comment[48]: MUSICBRAINZ_TRACKID=
    comment[49]: MEDIA=
    comment[50]: ORIGINALDATE=0000
    comment[51]: R128_ALBUM_GAIN=0
    comment[52]: R128_TRACK_GAIN=0
    comment[53]: REPLAYGAIN_ALBUM_GAIN=-14.51 dB
    comment[54]: REPLAYGAIN_ALBUM_PEAK=1.000031
    comment[55]: REPLAYGAIN_TRACK_GAIN=-10.03 dB
    comment[56]: REPLAYGAIN_TRACK_PEAK=0.875668
    comment[57]: SCRIPT=
    comment[58]: TITLE=Once upon a troubadour
    comment[59]: TRACK=13
    comment[60]: TRACKNUMBER=13
    comment[61]: TRACKTOTAL=13
    comment[62]: TRACKC=13
    comment[63]: TOTALTRACKS=13
METADATA block #3
  type: 1 (PADDING)
  is last: true
  length: 3549
```

**Comment by adrian-bl** [comment]
Hmm, that's interesting, haven't seen this yet. Maybe there is some odd splitting going on between ogg pages.

Could you send me this file to adrian@blinkenlights.ch?

**Comment by breversa** [comment]
Sure !
What file(s) do you want ? The FLAC, Ogg, or MP3 one(s) ?

**Comment by adrian-bl** [comment]
The Ogg and the MP3 would be the most interesting (the flac file probably has the same issue as the ogg file)

**Comment by adrian-bl** [comment]
Thanks for the sample files:

The values we read from the MP3 file seem to be correct: VLC reads exactly the same values.
![image](https://user-images.githubusercontent.com/568734/33574292-def19918-d938-11e7-8dbb-314c1907ed84.png)

For the ogg file: I also think that we are right there: Your file contains both `REPLAYGAIN_*_GAIN` and `R128_*_GAIN` tags:

![image](https://user-images.githubusercontent.com/568734/33574377-26791c8e-d939-11e7-9856-ae82d35523d3.png)

Vanilla music prefers R128 tags (if available, which is the case with your sample files) and converts the R128-gain values into 'legacy' values (and a R128-gain value of 0 happens to be 5dB).

The related code is here:
https://github.com/vanilla-music/vanilla/blob/master/src/ch/blinkenlights/android/vanilla/BastpUtil.java#L83

tl;dr: SoundKonverter seems to write invalid R128 tags.

**Comment by adrian-bl** [comment]
Interesting: did you use both bs1770gain and beets on these files or did beets write both tag types?

**Comment by breversa** [comment]
I'm new to Beets, but I used the Beets ReplayGain plugin to calculate the gain : http://docs.beets.io/en/v1.3.15/plugins/replaygain.html
Not sure if the R128 tags are Beets' or bs1770gain's work.

**Comment by adrian-bl** [comment]
Thanks for the info.

I'm closing this as i think we are doing the right thing in such a situation.
We may revisit our R128-tags support if they go out of fashion (i haven't seen them used that often in the wild)

**Comment by breversa** [comment]
I've tried to get rid of those R128 tags in Beets, but to no avail… so yeah, applying R128 tags only if ReplayGain ones don't exist may be a great idea in that case.

**Comment by breversa** [comment]
FWIW, I've created a bug for Beets here : https://github.com/beetbox/beets/issues/2757

**Comment by breversa** [comment]
@adrian-bl :
This issue is being worked on, but I have no idea if or when it'll be solved.

In the mean time, would you consider adding an option to Vanilla Music to pick which tag should have precedence, between ReplayGain or R128 ?
